packages/react/src/hooks/useConnectorClient.ts

```diff
@@ -81,7 +81,7 @@ export function useConnectorClient<
     chainId: parameters.chainId ?? chainId,
     connector: parameters.connector ?? connector,
   })
-  const enabled = Boolean(status !== 'disconnected' && (query.enabled ?? true))
+  const enabled = Boolean(status === 'connected' && (query.enabled ?? true))
```

Comment:
Agree with filtering out `'connecting'`. Let's add `'reconnecting'` here though as it enables optimistic updates.